Makefile

```diff
@@ -1,19 +1,30 @@
-.PHONY: image composer-update citest test
+.PHONY: image composer-update test
 
 IMAGE_NAME ?= codeclimate/codeclimate-phpmd
 
 image:
 	docker build --tag $(IMAGE_NAME) .
 
+test-image: image
+	docker build \
+		--build-arg BASE_IMAGE=${IMAGE_NAME} \
+		--tag $(IMAGE_NAME)-test \
+		--file Dockerfile.test .
+
 composer-update:
 	docker run \
 	  --rm \
 	  --volume $(PWD)/composer.json:/usr/src/app/composer.json:ro \
+	  --volume $(PWD)/composer.lock:/usr/src/app/composer.lock \
 	  $(IMAGE_NAME) \
-	  sh -c 'cd /usr/src/app && composer update && cat composer.lock' > composer.lock
+	  sh -c 'cd /usr/src/app && composer update'
 
-citest:
-	docker run --rm $(IMAGE_NAME) sh -c "cd /usr/src/app && vendor/bin/phpunit --bootstrap engine.php ./tests"
+citest: test
 
-test: image
-	docker run --rm $(IMAGE_NAME) sh -c "cd /usr/src/app && vendor/bin/phpunit --bootstrap engine.php ./tests"
+test:
+	@$(MAKE) test-image > /dev/null
```

Comment:
is the step `Prebuild image` necessary, if you have `make test-image` here?

Reply:
It's not necessary but it makes the test step on CI less noisy.

Reply:
ohh got it! that's ok